sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_do_while.py

```diff
@@ -54,4 +55,29 @@ def pipeline_with_do_while_node():
         _from_component_func=True,
     )
     do_while_node.set_limits(max_iteration_count=max_iteration_count)
+
+    def _infer_and_update_body_input_from_mapping():
+        # pylint: disable=protected-access
+        for output_name, body_input in mapping.items():
+            # handle case that mapping key is a NodeOutput
+            if isinstance(output_name, NodeOutput):
+                output_name = output_name.port_name
```

Comment:
Not sure why we should keeping `port_name` public. @wangchao1230 maybe we should make it private

Reply:
Updated this to `_port_name`